broadcast/src/buffered/engine.rs

```diff
@@ -57,28 +58,29 @@ pub struct Engine<
     // Messaging
     ////////////////////////////////////////
     /// The mailbox for receiving messages.
-    mailbox_receiver: mpsc::Receiver<Message<P, D, M>>,
+    mailbox_receiver: mpsc::Receiver<Message<P, Di, Dd, M>>,
 
     /// Pending requests from the application.
-    waiters: HashMap<D, Vec<oneshot::Sender<M>>>,
+    #[allow(clippy::type_complexity)]
+    waiters: HashMap<Di, Vec<(Option<P>, Option<Dd>, oneshot::Sender<M>)>>,
```

Comment:
Making this a standalone type is longer (because we have to redefine all the traits).

Reply:
I would prefer a type defined at the top of the file rather than a tuple here. While larger tuples can be fine in small pieces of code, seeing the use of `waiter.2` on a line that is hundreds of lines below this definition is hard to read

Reply:
Ah, I was being dumb (was just going to a tuple alias). Agree a struct is better.